jolt-core/src/jolt/instruction/virtual_assert_valid_div0.rs

```diff
@@ -0,0 +1,117 @@
+use crate::{
+    field::JoltField,
+    jolt::subtable::{left_is_zero::LeftIsZeroSubtable, right_is_ones::RightIsOnesSubtable},
+};
+use rand::prelude::StdRng;
+use rand::RngCore;
+use serde::{Deserialize, Serialize};
+
+use super::JoltInstruction;
+use crate::{
+    jolt::{instruction::SubtableIndices, subtable::LassoSubtable},
+    utils::instruction_utils::chunk_and_concatenate_operands,
+};
+
+#[derive(Copy, Clone, Default, Debug, Serialize, Deserialize)]
+/// (divisor, quotient)
+pub struct AssertValidDiv0Instruction<const WORD_SIZE: usize>(pub u64, pub u64);
+
+impl<const WORD_SIZE: usize> JoltInstruction for AssertValidDiv0Instruction<WORD_SIZE> {
+    fn operands(&self) -> (u64, u64) {
+        (self.0, self.1)
+    }
+
+    fn combine_lookups<F: JoltField>(&self, vals: &[F], C: usize, M: usize) -> F {
+        let vals_by_subtable = self.slice_values(vals, C, M);
+        let divisor_is_zero: F = vals_by_subtable[0].iter().product();
+        let quotient_is_ones: F = vals_by_subtable[1].iter().product();
+
+        F::one() - divisor_is_zero + divisor_is_zero * quotient_is_ones
+    }
+
+    fn g_poly_degree(&self, C: usize) -> usize {
+        C
```

Comment:
Should this be 2xC? `degree(divisor_is_zero) = C` `degree(quotient_is_ones) = C`? 

If so, I think this increases our primary sumcheck degree to (C+1)+2 -> 2C+2.

Probably nothing to be done about it but worth noting.

Reply:
ah actually I see a way to keep this at C by changing the `RIghtIsOnesSubtable`